File: app/src/main/java/khorunzhyicom/alex/theposter/di/modules/AppModule.java
package khorunzhyicom.alex.theposter.di.modules;

import android.content.Context;
import android.content.res.Resources;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import khorunzhyicom.alex.theposter.App;

@Module
public class AppModule {

    protected App app;

    public AppModule(App app) {
        this.app = app;
    }

    @Provides @Singleton Context provideContext() {
        return app.getApplicationContext();
    }

    @Provides @Singleton App provideApplication() {
        return app;
    }

    @Provides @Singleton Resources provideResources() {
        return app.getResources();
    }
}

File: app/src/main/java/khorunzhyicom/alex/theposter/di/modules/AppModule.kt
package khorunzhyicom.alex.theposter.di.modules

import android.content.Context
import android.content.res.Resources
import dagger.Module
import dagger.Provides
import khorunzhyicom.alex.theposter.App
import javax.inject.Singleton

@Module
class AppModule(var app: App) {

    //@Provides @Singleton fun provideContext(): Context = app.applicationContext

    //@Provides @Singleton fun provideResources(): Resources = app.resources
}
